### PR TITLE
[query] Perform a shufflerectomy (remove the shuffle service and code)

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: dev-docs/hail-query-lifecycle.md [hunk 1]
```diff
@@ -73,15 +73,8 @@ When invoked from Python, the JVM backends perform the following:
      CollectDistributedArray IR which represents the execution of a
      stage of the pipeline.  See `LowerTableIR`.
    - Shuffles (distributed sorts) are implemented on a per-backend
```
Comment (lgruen) [inline dev-docs/hail-query-lifecycle.md:75]: Very interesting! Is there already a design doc?

Comment (tpoterba) [reply dev-docs/hail-query-lifecycle.md:75]: Notes from our first (only) meeting here:
https://gist.github.com/tpoterba/cd590b70ec303f23353b4dc4c7fdfa7d

**Comment by daniel-goldstein** [comment]
As far as I'm aware, resolving the shuffler is the only thing `address` is doing at the moment. Is there a reason to keep that service around once this is removed?

**Comment by danking** [comment]
No, let's get rid of it too after this merges.

**Review by danking — APPROVED** [review]

**Pull request merged**